tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -3095,110 +3091,206 @@ LogicalResult ConvertTFLFakeQuantOp::matchAndRewrite(
+     }                                                                         \
+     auto it##PAT = legalization_enable.find(#PAT);                            \
+     if (it##PAT != legalization_enable.end())                                 \
+     {                                                                         \
+       if (it##PAT->second)                                                    \
+         DEF_PATTERN_INSERT(PAT);                                              \
+     }                                                                         \
+     if (it##PAT == legalization_enable.end())                                 \
+       DEF_PATTERN_INSERT(PAT);
+
+  #define DEF_FUNCTION_UNARY(PAT)                                              \
+     if (legalization_enable.size() == 0) {                                    \
+       DEF_PATTERN_INSERT_UNARY(PAT);                                          \
+     }                                                                         \
+     auto it##PAT = legalization_enable.find(#PAT);                            \
+     if (it##PAT != legalization_enable.end())                                 \
+     {                                                                         \
+       if (it##PAT->second)                                                    \
+         DEF_PATTERN_INSERT_UNARY(PAT);                                        \
+     }                                                                         \
+     if (it##PAT == legalization_enable.end())                                 \
+       DEF_PATTERN_INSERT_UNARY(PAT);
+
+  #define DEF_FUNCTION_BINARY(PAT)                                             \
+     if (legalization_enable.size() == 0) {                                    \
+       DEF_PATTERN_INSERT_BINARY(PAT);                                         \
+     }                                                                         \
+     auto it##PAT = legalization_enable.find(#PAT);                            \
+     if (it##PAT != legalization_enable.end())                                 \
+     {                                                                         \
+       if (it##PAT->second)                                                    \
+         DEF_PATTERN_INSERT_BINARY(PAT);                                       \
+     }                                                                         \
+     if (it##PAT == legalization_enable.end())                                 \
+       DEF_PATTERN_INSERT_BINARY(PAT);
+
+  DEF_FUNCTION_UNARY(Abs);
+  DEF_FUNCTION_UNARY(Ceil);
+  DEF_FUNCTION_UNARY(Floor);
+  DEF_FUNCTION_UNARY(Exp);
+  DEF_FUNCTION_UNARY(Log);
+  DEF_FUNCTION_UNARY(Rsqrt);
+  DEF_FUNCTION_UNARY(LogicalNot);
+  DEF_FUNCTION_UNARY(Cast);
+
   populateWithGenerated(patterns);
 
-#define DEF_PATTERN_INSERT(PAT) patterns.insert<Convert##PAT##Op>(ctx);
-  DEF_PATTERN_INSERT(TFLRelu);
```

Comment:
Do not change the name of the macro used for insertion. It generates a huge amount of diff spam.

Reply:
Apologies but not sure I follow this comment.

Reply:
Done